Fix environment variables not taken into account

FlorentFlament opened the pull request.

felixb commented:
I don't see, what is fixed with your commit.
Please give an example of old and new behavior.

FlorentFlament commented:
As far as I could observe, environment variables are not taken into account by the CLI in the current implementation:

    florent@critnux:~/work/marathon-api [fix_env]$ echo $MARATHON_USER
    f.flament
    florent@critnux:~/work/marathon-api [fix_env]$ echo $MARATHON_PASSWORD
    myveryverysecurepassword
    florent@critnux:~/work/marathon-api [fix_env]$ echo $MARATHON_INSECURE
    true
    florent@critnux:~/work/marathon-api [fix_env]$ git checkout upstream
    Switched to branch 'upstream'
    Your branch is up-to-date with 'origin/upstream'.
    florent@critnux:~/work/marathon-api [upstream]$ marathon list_tasks -i /hello
    /opt/chefdk/embedded/lib/ruby/2.1.0/net/http.rb:920:in `connect': SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed (OpenSSL::SSL::SSLError)
            from /opt/chefdk/embedded/lib/ruby/2.1.0/net/http.rb:920:in `block in connect'
            from /opt/chefdk/embedded/lib/ruby/2.1.0/timeout.rb:91:in `block in timeout'
            from /opt/chefdk/embedded/lib/ruby/2.1.0/timeout.rb:101:in `call'
            from /opt/chefdk/embedded/lib/ruby/2.1.0/timeout.rb:101:in `timeout'
            from /opt/chefdk/embedded/lib/ruby/2.1.0/net/http.rb:920:in `connect'
            from /opt/chefdk/embedded/lib/ruby/2.1.0/net/http.rb:863:in `do_start'
            from /opt/chefdk/embedded/lib/ruby/2.1.0/net/http.rb:852:in `start'
            from /opt/chefdk/embedded/lib/ruby/2.1.0/net/http.rb:1369:in `request'
            from /home/florent/.chefdk/gem/ruby/2.1.0/gems/httparty-0.13.5/lib/httparty/request.rb:101:in `perform'
            from /home/florent/.chefdk/gem/ruby/2.1.0/gems/httparty-0.13.5/lib/httparty.rb:522:in `perform_request'
            from /home/florent/.chefdk/gem/ruby/2.1.0/gems/httparty-0.13.5/lib/httparty.rb:460:in `get'
            from /home/florent/work/marathon-api/lib/marathon/connection.rb:105:in `request'
            from /home/florent/work/marathon-api/lib/marathon/connection.rb:43:in `block (2 levels) in <class:Connection>'
            from /home/florent/work/marathon-api/lib/marathon/task.rb:95:in `get'
            from /home/florent/work/marathon-api/lib/marathon/task.rb:50:in `get'
            from /home/florent/work/marathon-api/bin/marathon:103:in `subcmd_list_tasks'
            from /home/florent/work/marathon-api/bin/marathon:236:in `run_subcmd'
            from /home/florent/work/marathon-api/bin/marathon:250:in `<top (required)>'
            from /home/florent/.chefdk/gem/ruby/2.1.0/bin/marathon:23:in `load'
            from /home/florent/.chefdk/gem/ruby/2.1.0/bin/marathon:23:in `<main>'
    florent@critnux:~/work/marathon-api [upstream]$ marathon -I list_tasks -i /hello
    /home/florent/work/marathon-api/lib/marathon/connection.rb:93:in `parse_response': <!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN"> (Marathon::Error::UnexpectedResponseError)
    <html><head>
    <title>401 Authorization Required</title>
    </head><body>
    <h1>Authorization Required</h1>
    <p>This server could not verify that you
    are authorized to access the document
    requested.  Either you supplied the wrong
    credentials (e.g., bad password), or your
    browser doesn't understand how to supply
    the credentials required.</p>
    <hr>
    <address>Apache Server at marathon-par.central.criteo.preprod Port 443</address>
    </body></html>
            from /home/florent/work/marathon-api/lib/marathon/connection.rb:105:in `request'
            from /home/florent/work/marathon-api/lib/marathon/connection.rb:43:in `block (2 levels) in <class:Connection>'
            from /home/florent/work/marathon-api/lib/marathon/task.rb:95:in `get'
            from /home/florent/work/marathon-api/lib/marathon/task.rb:50:in `get'
            from /home/florent/work/marathon-api/bin/marathon:103:in `subcmd_list_tasks'
            from /home/florent/work/marathon-api/bin/marathon:236:in `run_subcmd'
            from /home/florent/work/marathon-api/bin/marathon:250:in `<top (required)>'
            from /home/florent/.chefdk/gem/ruby/2.1.0/bin/marathon:23:in `load'
            from /home/florent/.chefdk/gem/ruby/2.1.0/bin/marathon:23:in `<main>'
    florent@critnux:~/work/marathon-api [upstream]$ git checkout fix_env
    Switched to branch 'fix_env'
    Your branch is up-to-date with 'origin/fix_env'.
    florent@critnux:~/work/marathon-api [fix_env]$ marathon list_tasks -i /hello
    Task ID:    hello.7ea7fc20-57ba-11e5-a466-1458d05b0380
    App ID:     /hello
    Host:       mesos-slave002-par.central.criteo.preprod
    Ports:      31445
    Staged at:  2015-09-10T12:50:20.515Z
    Started at: 2015-09-10T12:50:20.889Z
    Version:    2015-09-10T12:50:16.848Z

    Task ID:    hello.7e0f3f8f-57ba-11e5-a466-1458d05b0380
    App ID:     /hello
    Host:       mesos-slave01-par.central.criteo.preprod
    Ports:      31596
    Staged at:  2015-09-10T12:50:19.514Z
    Started at: 2015-09-10T12:50:20.132Z
    Version:    2015-09-10T12:50:16.848Z

    florent@critnux:~/work/marathon-api [fix_env]$

felixb commented:
This was actually a bug in the CLI handling default options.
